code/src/component/GameContainer.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Typewriter } from 'react-typewriting-effect';
+import 'react-typewriting-effect/dist/index.css';
+import Zoom from 'react-reveal/Zoom';
+
+import { GameStartContainer } from './GameStartContainer'
+import { game } from '../reducers/game';
+import { generateNewDirection } from '../reducers/reusable';
+import { generateGameStart } from '../reducers/reusable';
+import { Wrapper } from '../styling/typewriter';
+import { GameScreen, DescriptionText, ButtonsContainer, MoveButton } from '../styling/gamecontainer';
+import Compass from '../img/compass.png';
+
+export const GameContainer = () => {
+    const dispatch = useDispatch();
+    const [actionDescription, setActionDescription] = useState('');
+
+    // Get from store(redux) 
+    const gameStore = useSelector(store => store.game);
+    const username = useSelector(store => store.game.username);
+    const description = useSelector(store => store.game.description);
+    const fetching = useSelector(store => store.game.fetching);
+    const timing = useSelector(store => store.game.timer);
+    
+    const onGameGenerate = () => {
+        // Runs the fetch function generateGameStart from reusable
+        dispatch(generateGameStart());
+    }
+
+    // onGameGenerate renders when username is updated
+    useEffect(onGameGenerate, [username])
+
+    const onGameMoveUpdate = (action) => {
+        setActionDescription(action.description)
+        onTimer();
```

Comment:
It as really nice to have the description as loader, maybe you could have had the timer set for a second or two longer, when the description was long it was a bit stressful to keep up reading